packages/test-utils/src/reporter.ts

```diff
@@ -37,31 +41,37 @@ interface TestRunTestInputModel {
   recordingIds: string[];
 }
 
-export interface TestIdContext {
-  title: string;
-  scope: string[];
-  attempt: number;
-}
-
 export type UploadStatusThreshold = "all" | "failed-and-flaky" | "failed";
 
 type UploadStatusThresholdInternal = UploadStatusThreshold | "none";
 
 export type UploadOption =
   | boolean
   | {
+      /**
+       * It minimizes uploads related to a single test within a shard
+       */
```

Comment:
I think it might be useful to be slight more verbose here, e.g.

```suggestion
      /**
       * Minimize the number of recordings uploaded for a test attempt (within a shard).
       * e.g. Only one recording would be uploaded for a failing test attempt, regardless of retries.
       * e.g. Two recordings would be uploaded for a flaky test attempt (the passing test and one of the failures).
       */
```